common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -57,3 +64,167 @@ def perform_destroy(self, instance):
             )
         if save_model:
             instance.save()
+
+
+class EntitlementEnrollmentViewSet(viewsets.GenericViewSet):
+    """
+    Endpoint in the Entitlement API to handle the Enrollment of a User's Entitlement.
+    This API will handle
+        - Enroll
+        - Unenroll
+        - Switch Enrollment
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = CourseEntitlementSerializer
+
+    def _verify_course_run_for_entitlement(self, entitlement, course_run_id):
+        """
+        Verifies that a Course run is a child of the Course assigned to the entitlement.
+        """
+        course_run_valid = False
+        course_runs = get_course_runs_for_course(entitlement.course_uuid)
+        for run in course_runs:
+            if course_run_id == run.get('key', ''):
+                course_run_valid = True
+                break
```

Comment:
Nit: Personally I hate breaks and think they're icky, since we're not doing anything with the course_run_id we could swap this to 

```
for run in course_runs:
    if course_run_id == run.get('key', ''):
        return True
return False
```